docs/notebooks/jax-101/01-jax-basics.md

```diff
@@ -0,0 +1,375 @@
+
+We will explain other places where the JAX idiosyncracies become relevant as they come up. There is even a section that focuses entirely on getting used to the functional programming style of handling state: [Part 7: Problem of State](https://colab.research.google.com/github/google/jax/blob/master/docs/notebooks/jax-101/07-state.ipynb). However, if you're impatient, you can find a [summary of JAX's sharp edges](https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html) in the JAX docs.
+
++++ {"id": "dFn_VBFFlGCz"}
+
+## Your first JAX training loop
+
+We still have much to learn about JAX, but you already know enough to understand how we can use JAX to build a simple training loop.
+
+To keep things simple, we'll start with a linear regression. 
+
+Our data is sampled according to $y = w_{true} x + b_{true} + \epsilon$.
+
+```{code-cell} ipython3
+:id: WGgyEWFqrPq1
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+xs = np.random.normal(size=(100,))
+noise = np.random.normal(scale=0.1, size=(100,))
+ys = xs * 3 - 1 + noise
+
+plt.scatter(xs, ys)
+```
+
++++ {"id": "RTh22mo4rR1x"}
+
+Therefore, our model is $\hat y(x; \theta) = wx + b$.
+
+We will use a single array, `theta = [w, b]` to house both parameters:
+
+```{code-cell} ipython3
+:id: TnVrRTMamyzb
+
+def model(theta, x):
+  """Computes wx + b on a batch of input x."""
+  w, b = theta
+  return w * x + b
+```
+
++++ {"id": "qCrLmmKrn9_h"}
+
+The loss function is $J(x, y; \theta) = (\hat y - y)^2$.
+
+```{code-cell} ipython3
+:id: 07eMcDLMn9Ww
+
+def loss_fn(theta, x, y):
+  prediction = model(theta, x)
+  return jnp.mean((prediction-y)**2)
+```
+
++++ {"id": "ejMt4dulnoYX"}
+
+How do we optimize a loss function? Using gradient descent. At each update step, we will find the gradient of the loss w.r.t. the parameters, and take a small step in the direction of steepest descent:
+
+$\theta_{new} = \theta - 0.1 (\nabla_\theta J) (x, y; \theta)$
+
+```{code-cell} ipython3
+:id: 2I6T5Wphpaaa
+
+def update(theta, x, y, lr=0.1):
+  return theta - lr * jax.grad(loss_fn)(theta, x, y)
+```
+
++++ {"id": "MAUL1gT_opVn"}
+
+In JAX, it's common to define an `update()` function that is called every step, taking the current parameters as input and returning the new parameters. This is a natural consequence of JAX's functional nature, and is explained in more detail in [The Problem of State](https://colab.research.google.com/github/google/jax/blob/master/docs/notebooks/jax-101/07-state.ipynb).
+
+This function can then be jit-compiled in its entirety for maximum efficiency. The next guide will explain exactly how `jax.jit` works, but if you want to, you can try adding `@jax.jit` before the `update()` definition, and see how the training loop below runs much faster.
```

Comment:
Tiny suggestion:

- write the just-in-time acronym as `JIT` instead of `jit` (similar to @gnecula 's https://github.com/google/jax/pull/5833/files#diff-12d1ee2b456267cd784f8859ceb40dd48e54bb5c77da97319143fae2bee2e48aR4-R8)

```suggestion
This function can then be JIT-compiled in its entirety for maximum efficiency. The next guide will explain exactly how `jax.jit` works, but if you want to, you can try adding `@jax.jit` before the `update()` definition, and see how the training loop below runs much faster.
```

Reply:
Done in the most recent commit - thanks!